pkg/reconciler/managed/reconciler.go

```diff
@@ -792,6 +795,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	if meta.WasDeleted(managed) {
 		log = log.WithValues("deletion-timestamp", managed.GetDeletionTimestamp())
 
+		if len(managed.GetFinalizers()) > 1 {
+			log.Debug("Waiting for additional finalizers to be removed")
```

Comment:
Similar to my previous comment above, logging the number of remaining finalizers would be useful here.

Reply:
This code was removed as agreed.